pkg/ext/stores/useractivity/store.go

```diff
@@ -0,0 +1,307 @@
+	if activityToken.Name != objUserActivity.Name {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("token name mismatch: have %s - expected %s", activityToken.Name, objUserActivity.Name))
+	}
+
+	// once validated the request, we can define the lastActivity time.
+	newIdleTimeout := metav1.Time{
+		Time: lastActivity.Add(time.Minute * time.Duration(idleTimeout)).UTC(),
+	}
+	objUserActivity.Status.ExpiresAt = newIdleTimeout.String()
+
+	// if it's not a dry-run, commit the changes
+	if !dryRun {
+		patch, err := json.Marshal([]struct {
+			Op    string `json:"op"`
+			Path  string `json:"path"`
+			Value any    `json:"value"`
+		}{{
+			Op:    "replace",
+			Path:  "/activityLastSeenAt",
+			Value: newIdleTimeout,
+		}})
+		if err != nil {
+			return nil, apierrors.NewInternalError(fmt.Errorf("%w", err))
+		}
+		_, err = s.tokens.Patch(activityToken.GetName(), types.JSONPatchType, patch)
+		if err != nil {
+			return nil, apierrors.NewInternalError(fmt.Errorf("failed to patch token: %w", err))
+		}
+	}
+
+	return objUserActivity, nil
+}
+
+// Get implements [rest.Getter]
+// Get returns the UserActivity based on the token name.
+// It is used to know, from the frontend, how much time
+// remains before the idle timeout triggers.
+func (s *Store) Get(ctx context.Context,
+	name string,
+	options *metav1.GetOptions) (runtime.Object, error) {
+	userInfo, err := s.userFrom(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	extras := userInfo.GetExtra()
+
+	authTokenID := first(extras[common.ExtraRequestTokenID])
+	if authTokenID == "" {
+		return nil, apierrors.NewForbidden(GVR.GroupResource(), "", fmt.Errorf("missing request token ID"))
+	}
+
+	// retrieve auth token
+	authToken, err := s.tokenCache.Get(authTokenID)
+	if err != nil {
+		return nil, apierrors.NewForbidden(GVR.GroupResource(), "", fmt.Errorf("error getting request token %s: %w", authTokenID, err))
+	}
+
+	// retrieve activity token
+	activityToken, err := s.tokenCache.Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, apierrors.NewBadRequest(fmt.Sprintf("token not found %s: %v", name, err))
+		} else {
+			return nil, apierrors.NewInternalError(fmt.Errorf("failed to get token %s: %w", name, err))
+		}
+	}
+
+	// verify auth and activity token has the same auth provider
+	if authToken.AuthProvider != activityToken.AuthProvider {
+		return nil, apierrors.NewForbidden(GVR.GroupResource(), "", fmt.Errorf("request token %s and activity token %s have different auth providers", authTokenID, name))
+	}
+
+	// verify auth and activity token has the same auth user principal
+	if authToken.UserPrincipal.Name != activityToken.UserPrincipal.Name {
+		return nil, apierrors.NewForbidden(GVR.GroupResource(), "", fmt.Errorf("request token %s and activity token %s have different user principals", authTokenID, name))
+	}
+
+	// crafting UserActivity from requested Token name.
+	ua := &ext.UserActivity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
```

Comment:
This is under assumption that we equate useractivity and its backing session token.
```suggestion
			CreationTimestamp: activityToken.CreationTimestamp,
			Name:              name,
```
Without this you'll get
```sh
kubectl get useractivity token-xxxxx
NAME          AGE
token-xxxxx   <unknown>
```